cmd/libsnap-confine-private/classic-test.c

```diff
@@ -20,56 +20,102 @@
 
 #include <glib.h>
 
-const char *os_release_classic = ""
+/* restore_os_release is an internal helper for mock_os_release */
+static void restore_os_release(gpointer * old)
+{
+	unlink(os_release);
+	os_release = (const char *)old;
+}
+
+/* mock_os_release replaces the contents of /etc/os-release as seen by classic.c
+   mocked may be NULL to have the code refer to an absent file. */
+static void mock_os_release(const char *mocked)
+{
+	const char *old = os_release;
+	if (mocked != NULL) {
+		os_release = "os-release.test";
+		g_file_set_contents(os_release, mocked, -1, NULL);
+	} else {
+		os_release = "os-release.missing";
+	}
+	g_test_queue_destroy((GDestroyNotify) restore_os_release,
+			     (gpointer) old);
+}
+
+/* restore_meta_snap_yaml is an internal helper for mock_meta_snap_yaml */
+static void restore_meta_snap_yaml(gpointer * old)
+{
+	unlink(meta_snap_yaml);
+	meta_snap_yaml = (const char *)old;
+}
+
+/* mock_meta_snap_yaml replaces the contents of /meta/snap.yaml
```

Comment:
I think this comment is slightly misleading in the light of what the function is actually doing. It makes meta_snap_yaml point to a fake file created as part of mocking (and this affects os detection). Can you fix/expand the comment?

Reply:
Done